File: dvbViewerController/src/main/java/org/dvbviewer/controller/data/timer/retrofit/TimerRequestBodyConverter.java
package org.dvbviewer.controller.data.timer.retrofit;

import org.dvbviewer.controller.entities.Timer;

import java.util.List;

import okhttp3.MediaType;
import okhttp3.RequestBody;
import retrofit2.Converter;

final class TimerRequestBodyConverter implements Converter<List<Timer>, RequestBody> {

    private static final MediaType MEDIA_TYPE = MediaType.parse("text");

    @Override public RequestBody convert(List<Timer> value) {
        return RequestBody.create(MEDIA_TYPE, value.toString());
    }

}

File: dvbViewerController/src/main/java/org/dvbviewer/controller/data/timer/retrofit/TimerRequestBodyConverter.kt
package org.dvbviewer.controller.data.timer.retrofit

import okhttp3.MediaType
import okhttp3.RequestBody
import org.dvbviewer.controller.data.entities.Timer
import retrofit2.Converter

internal class TimerRequestBodyConverter : Converter<List<Timer>, RequestBody> {

    override fun convert(value: List<Timer>): RequestBody {
        return RequestBody.create(MEDIA_TYPE, value.toString())
    }

    companion object {

        private val MEDIA_TYPE = MediaType.parse("text")
    }

}